src/Controller/Traits/DocumentTreeConfigTrait.php

```diff
@@ -16,21 +16,30 @@
 
 namespace Pimcore\Bundle\AdminBundle\Controller\Traits;
 
-use Pimcore\Bundle\AdminBundle\Event\ElementAdminStyleEvent;
-use Pimcore\Config;
-use Pimcore\Model\Document;
+use Pimcore\Bundle\AdminBundle\Service\ElementServiceInterface;
 use Pimcore\Model\Element\ElementInterface;
-use Pimcore\Model\Site;
-use Pimcore\Tool\Admin;
-use Pimcore\Tool\Frontend;
+use Symfony\Contracts\Service\Attribute\Required;
 
 /**
  * @internal
+ *
+ * @deprecated
  */
 trait DocumentTreeConfigTrait
 {
     use AdminStyleTrait;
 
+    protected ElementServiceInterface $elementService;
+
+    /**
+     * @param ElementServiceInterface $elementService
+     */
+    #[Required]
+    public function setElementService(ElementServiceInterface $elementService): void
```

Comment:
this looks like some leftover

Reply:
this is used here https://github.com/pimcore/admin-ui-classic-bundle/blob/051bec17723cea0171d282d48fd4af244734c352/src/Controller/Traits/DocumentTreeConfigTrait.php#L52

the service needs to be injected somehow so we use required attribute

Reply:
ahh, right, didn't recognize the setter injection